src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -64,4 +73,27 @@ public List<AdvertBasicInfoDTO> getByCategory(String category) {
         }
 
     }
+
+    @Transactional
+    public AdvertFullInfoDTO createAdvert(AdvertCreateDTO advertCreateDTO) {
+        Long id = advertCreateDTO.getUserCreatorId();
+        if (!userService.userExistsById(id)) {
+            throw new UserNotFoundException("User with id " + id + " does not exist.");
+        }
```

Comment:
Yeap, i would move `findById` (row #91-92) here instead of `userExistsById` (remove if-check)

Reply:
This logic, along with the userExistsById method, is redundant since `User user = userRepository.findById(id).orElseThrow()` already throws exception if user does not exist.

Reply:
yeah, for sure, thanks